File: 908-middle-of-the-linked-list/middle-of-the-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* middleNode(ListNode* head) {
        ListNode* current = head;
        int length = 0;
        while(current){
            length++;
            current = current->next;
        }
        current = head;
        int mid = length/2 + 1;
        for(int i = 1; i < mid; i++){
            current = current->next;
        }
        return current;
    }
};

File: 908-middle-of-the-linked-list/middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def middleNode(self, head):
       elements = []
       while head != None :
           elements.append(head)
           head = head.next
       return elements[(len(elements)/2)]